service/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -273,6 +276,41 @@ public List<UUID> listLandingZoneIds(BearerToken bearerToken, UUID billingProfil
         .toList();
   }
 
+  /**
+   * Lists landing zone record for a billing profile ID.
+   *
+   * @param bearerToken bearer token of the calling user.
+   * @param billingProfileId the billing profile ID to search.
+   * @return landing zone record.
+   */
+  public LandingZoneRecord getLandingZoneRecord(BearerToken bearerToken, UUID billingProfileId) {
+    // Call BPM to validate user access to billing profile.
+    bpmService.verifyUserAccess(bearerToken, billingProfileId);
+
+    var landingZoneRecord = getLandingZoneRecordByBillingProfile(bearerToken, billingProfileId);
+    return landingZoneRecord;
+  }
+
+  /**
+   * Lists landing zones for all billing profiles that the calling user has access to.
+   *
+   * @param bearerToken bearer token of the calling user.
+   * @return list of landing zone records.
+   */
+  public List<LandingZoneRecord> listLandingZoneRecords(BearerToken bearerToken) {
+    // Call BPM to get the landing zone target.
+    var profileList = bpmService.getBillingProfiles(bearerToken);
+    var landingZoneRecords = new ArrayList<LandingZoneRecord>();
+
+    // Query the database for landing zone records with the profile Ids.
+    profileList.getItems().stream()
+        .forEach(
+            profileModel ->
+                landingZoneRecords.add(
+                    getLandingZoneRecordByBillingProfile(bearerToken, profileModel.getId())));
```

Comment:
This is listing all billing-profiles a user has access to and iterating through them. But a given billing-profile might not have a landing zone. I think this code would throw LandingZoneNotFoundException in this case, which is probably not what we want.

Another approach would be to call Sam [listResourcesAndPolicies](https://sam.dsde-dev.broadinstitute.org/#/Resources/listResourcesAndPoliciesV2) to list all landing-zone IDs the caller has access to. Then you could query the LZ DB for each ID returned from Sam. This would avoid needing to call BPM altogether.

Reply:
Good point, will review this.